File: Assignment34/Program4.java

//3. Write a program which accept number from user and return the Multiplication of all the digit
import java.util.*;

class Program4
    
{ 
   public static int Mutiply(int iNo)
   {
         int iDigit = 0;
         int iMult = 1;

         while(iNo != 0)
         {
            iDigit = iNo % 10;

             if(iDigit == 0) //updator for zero
             {
                iDigit = 1; 
             }

            iMult = iMult *iDigit;
             iNo = iNo /10;
             
         }
         return iMult;
   } 

   public static void main(String Args[])
   {
       Scanner sobj = new Scanner(System.in);
       int iValue = 0;
       int iRet = 0;

       System.out.println("Enter Number:");
       iValue = sobj.nextInt();
       
       iRet = Mutiply(iValue);
       System.out.println(" "+iRet);

   }
}

File: Assignment50/PRogram4.c


//Write a recursive program which accept number and return smallest digit
#include<stdio.h>
//Need corruction
//Recursive function
int Min(int iNo)
{
      int iMin = 0;
      int iDigit = 0;

    if(iNo != 0)
    {
       iDigit = iNo % 10;
       if(iDigit <= iNo)  
       {
           iMin = iDigit;
       }

       iNo = iNo / 10;
       Min(iNo);//recursive call

    }
    return iMin;

}

int main()
{
  
    int iValue = 0, iRet =0;
    printf("Enter Number :");
    scanf("%d",&iValue);

    iRet = Min(iValue);
    printf("Min Digit is : %d ",iRet);
 
    return 0;
}